### PR TITLE
Disable scroll on body for <amp-sidebar>/<amp-lightbox>

### LINKED ISSUE #11225: Body Scrolling doesn't stop on amp-lightbox on iOS
If a lengthy page has an amp-lightbox extention in use and the lightbox is active, the scrollbar still shows and background main body could be scrolled while lightbox is in view for iOS devices.

PS: This issue was reported before too and was closed [#6487], but still occurs.

Platforms: Safari (iOS), Opera Mini (Android)

### PR DESCRIPTION
**Pull request by nainar**
On iOS the scroll handler will scroll the body element if the active element (`<amp-sidebar>`/`<amp-lightbox>`) isn't scrollable itself. This would be in the case of:
- A `<amp-sidebar>` element that doesn't have scroll (too few elements)
- Always for `<amp-lightbox>`

Solution taken from: https://stackoverflow.com/questions/9538868/prevent-body-from-scrolling-when-a-modal-is-opened

Fixes #11225

### CODE REVIEW

File: extensions/amp-lightbox/0.1/amp-lightbox.js [hunk 1]
```diff
@@ -231,6 +231,10 @@ class AmpLightbox extends AMP.BaseElement {
 
     this.getViewport().enterLightboxMode(this.element)
         .then(() => this.finalizeOpen_());
+    setImportantStyles(this.document_.body, {
```
Comment (cathyxz) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: Nit: add comment explaining why this is necessary.

File: src/service/viewport/viewport-impl.js [hunk 2]
```diff
@@ -782,6 +782,11 @@ export class Viewport {
   enterOverlayMode() {
     this.disableTouchZoom();
     this.disableScroll();
```
Comment (nainar) [inline src/service/viewport/viewport-impl.js]: This should be disabling scroll it sets overflow hidden but doesn't set the position to be fixed. Should move the code here.

File: src/service/viewport/viewport-binding-ios-embed-sd.js [hunk 3]
```diff
@@ -327,7 +327,10 @@ export class ViewportBindingIosEmbedShadowRoot_ {
   /** @override */
   disableScroll() {
     // TODO(jridgewell): Recursively disable scroll
-    setImportantStyles(this.scroller_, {'overflow-y': 'hidden'});
+    setImportantStyles(this.scroller_, {
+      'overflow-y': 'hidden',
+      'position': 'fixed',
```
Comment (aghassemi) [inline src/service/viewport/viewport-binding-ios-embed-sd.js]: oh, how does the `position` get reset to whatever it was before? looks like `resetScroll` needs updating as well.

Comment (aghassemi) [reply src/service/viewport/viewport-binding-ios-embed-sd.js]: I am also worried that setting `position:fixed` on the scroller may have side-effects as it changes the stacking context for all other elements including the "lightbox". I suggest this being behind an experiment flags that's slowly rolled into prod.

Comment (aghassemi) [reply src/service/viewport/viewport-binding-ios-embed-sd.js]: nevermind, oh realized this is only for the new iOS shadow-dom that @dvoytenko added.

**Pull request merged**